controllers/admin/AdminProductsController.php

```diff
@@ -866,20 +866,34 @@ public function processFeatures($id_product = null)
 
             // add new objects
             $languages = Language::getLanguages(false);
-            foreach ($_POST as $key => $val) {
-                if (preg_match('/^feature_([0-9]+)_value/i', $key, $match)) {
-                    if ($val) {
-                        $product->addFeaturesToDB($match[1], $val);
+            $old_id_feature = -1;
+            $counter = 0;
+            foreach ($_POST as $key => $value) {
+                if (preg_match('/^feature_([0-9]+)_([0-9]+|)_value/i', $key, $match)) {
+                    if ($old_id_feature != $match[1]) {
+                        $old_id_feature = $match[1];
+                        $counter = 0;
+                    }
+                    if ($value) {
+                        $product->addFeaturesToDB($match[1], $value);
                     } else {
                         if ($default_value = $this->checkFeatures($languages, $match[1])) {
-                            $id_value = $product->addFeaturesToDB($match[1], 0, 1);
-                            foreach ($languages as $language) {
-                                if ($cust = Tools::getValue('custom_'.$match[1].'_'.(int)$language['id_lang'])) {
-                                    $product->addFeaturesCustomToDB($id_value, (int)$language['id_lang'], $cust);
-                                } else {
-                                    $product->addFeaturesCustomToDB($id_value, (int)$language['id_lang'], $default_value);
+                            do {
+                                $id_value = $product->addFeaturesToDB($match[1], 0, 1);
+                                $has_next_custom = false;
+                                foreach ($languages as $language) {
+                                    if ($cust = Tools::getValue('custom_' . $match[1] . '_' . (int)$language['id_lang'] . '_' . $counter)) {
```

Comment:
Hello @fatmaBouchekoua 
Please, see your code and when you have finished your feature, try to minimize your code, 
try if this works: 

```php
foreach ($languages as $language) {
    $customFeature = Tools::getValue('custom_' . $match[1] . '_' . (int)$language['id_lang'] . '_' . $counter);
    $finalFeature = $customFeature ? $customFeature : $default_value;

    $product->addFeaturesCustomToDB($id_value, (int)$language['id_lang'], $finalFeature);

    $has_next_custom = (bool)Tools::getValue('custom_' . $match[1] . '_' . (int)$language['id_lang'] . '_' . ($counter + 1));
}
```

Be easy, 
```php 
$something = false;
//some logic
if ($somethingelse) {
    $something = true;
}

// so
$something = false
//some logic
$something = (bool) $somethingelse
```

It's really more complicated to see if you code is correct/good or not :(